packages/components/src/styles/components/copy/snippet.scss

```diff
@@ -49,32 +49,34 @@
   color: var(--token-color-foreground-primary);
   background-color: transparent;
 
+  .hds-copy-snippet__icon {
+    color: var(--token-color-foreground-action);
+  }
+
   &:hover,
   &.mock-hover {
     background-color: var(--token-color-surface-interactive);
     border-color: var(--token-color-border-strong);
+
+    // Change the color of the icon in hover state, except if the component has a status of success or error
+    .hds-copy-snippet__icon:not(
```

Comment:
I'm not sure if this is the cleanest solution, but it does prevent some repetition in declaring the hover states.

Reply:
Throws an error in the linter, so I'm thinking this is not the best approach.

Reply:
not necessarily. I think you have a linter different from what is used in the repo. I'll have a look on monday

Reply:
I reworked this a bit passed a comma-separated list of classes rather than chaining multiple `:not()` pseudo classes together which seems to satisfy the linter.

Reply:
[suggestion] instead of doing all these `not()` overrides, I would follow the same approach used for the `Copy::Button` which is to increase the specificity for the "success/error" state classes (and in this case, since it would be used twice, I would extract it to a local mixin):

```
@mixin hds-copy-snippet-success-error() {
  &.hds-copy-snippet--status-success {
    background-color: var(--token-color-surface-interactive);

    .hds-copy-snippet__icon {
      color: var(--token-color-foreground-success);
    }
  }

  &.hds-copy-snippet--status-error {
    background-color: var(--token-color-surface-interactive);

    .hds-copy-snippet__icon {
      color: var(--token-color-foreground-critical);
    }
  }
}

.hds-copy-snippet {
  @include hds-focus-ring-with-pseudo-element();

  display: flex;
  gap: 8px;
  align-items: center;
  justify-content: space-between;
  padding: 6px 4px;
  text-align: left;
  border: 1px solid transparent;
  border-radius: 5px;
  cursor: pointer;
}

.hds-copy-snippet--color-primary {
  color: var(--token-color-foreground-action);
  background-color: transparent;

  &:hover,
  &.mock-hover {
    color: var(--token-color-foreground-action-hover);
    background-color: var(--token-color-surface-interactive);
    border-color: var(--token-color-border-strong);
  }

  &:active,
  &.mock-active {
    color: var(--token-color-foreground-action-active);
    background-color: var(--token-color-surface-interactive-active);
    border-color: var(--token-color-border-strong);
  }

  @include hds-copy-snippet-success-error();
}

.hds-copy-snippet--color-secondary {
  color: var(--token-color-foreground-primary);
  background-color: transparent;

  .hds-copy-snippet__icon {
    color: var(--token-color-foreground-action);
  }

  &:hover,
  &.mock-hover {
    background-color: var(--token-color-surface-interactive);
    border-color: var(--token-color-border-strong);

    .hds-copy-snippet__icon {
      color: var(--token-color-foreground-action-hover);
    }
  }

  &:active,
  &.mock-active {
    background-color: var(--token-color-surface-interactive-active);
    border-color: var(--token-color-border-strong);

    .hds-copy-snippet__icon {
      color: var(--token-color-foreground-action-active);
    }
  }

  @include hds-copy-snippet-success-error();
}

```

Reply:
This is a great suggestion, much cleaner!